src/bin/cobalt/new.rs

```diff
@@ -417,8 +443,17 @@ pub fn publish_document(file: &path::Path) -> Result<()> {
     let doc =
         cobalt_model::DocumentBuilder::<cobalt_model::FrontmatterBuilder>::new(front, content);
     let doc = doc.to_string();
-
     cobalt_model::files::write_document_file(doc, file)?;
-
+    if config.posts.publish_date_in_filename {
+        let posts = config.posts.clone().build()?;
+        if posts.pages.includes_file(&file) {
```

Comment:
This feels backwards to me.  We should be checking what collection it is in before doing the collection-specific processing.

Yes, it causes the extra clones.  When I get to the mythical day of switching to an ECS, that should be fixed.